components/brave_news/browser/brave_news_controller.cc

```diff
@@ -455,39 +458,85 @@ void BraveNewsController::GetImageData(const GURL& padded_image_url,
   const std::string ending = ".pad";
   if (file_name.length() >= file_name.max_size() - 1 ||
       file_name.length() <= ending.length()) {
-    std::optional<std::vector<uint8_t>> args;
-    std::move(callback).Run(std::move(args));
+    std::move(callback).Run(std::nullopt);
     return;
   }
   const bool is_padded =
       (file_name.compare(file_name.length() - ending.length(), ending.length(),
                          ending) == 0);
   VLOG(3) << "is padded: " << is_padded;
+  // Ensure we don't de-pad non PCDN images
+  if (is_padded && !padded_image_url.DomainIs(GetMatchingPCDNHostname())) {
+    DLOG(ERROR) << "Encountered a padded URL apparently not from PCDN: "
+                << padded_image_url.spec();
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
   // Make the request
   private_cdn_request_helper_.DownloadToString(
       padded_image_url,
+      base::BindOnce(&BraveNewsController::OnGetImageDataFetched,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback),
+                     is_padded));
+}
+
+void BraveNewsController::OnGetImageDataFetched(GetImageDataCallback callback,
+                                                const bool is_padded,
+                                                const int response_code,
+                                                const std::string& body) {
+  // Handle the response
+  VLOG(3) << "getimagedata response code: " << response_code;
+  // Attempt to remove byte padding if applicable
+  std::string_view body_payload(body.data(), body.size());
+  if (response_code < 200 || response_code >= 300 ||
```

Comment:
This can be simplified to:
```suggestion
  if (response_code != 200 ||
```

Reply:
Surely `201` is fine too? (and possibly `206`)`

Reply:
I don't think `201` (_Document created, URL follows_) would make sense and `206` (_Partial content follows_) would likely not work either. But thinking more about it, `203` (_Request fulfilled from cache_) would be okay too.

My thinking was that most `2xx` codes will mean the image is garbage (e.g. `204` and `205`) and in 95%+ of cases this will be a clean `200` when an image is returned.

Reply:
I just moved this code, but appreciated

Reply:
...and changed to `200` only